contracts/governance/FeeSharingCollector/FeeSharingCollector.sol

```diff
@@ -393,133 +422,261 @@ contract FeeSharingCollector is
+    ) external nonReentrant {
+        validRBTCBasedTokens(_tokens);
+
+        if (_receiver == ZERO_ADDRESS) {
+            _receiver = msg.sender;
+        }
+
+        uint256 tempRbtcAmountToSend;
+
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            if (_maxCheckpoints == 0) break;
+            address _token = _tokens[i];
+            uint256 startingCheckpoint = processedCheckpoints[msg.sender][_token];
+
+            (uint256 totalAmount, uint256 endToken) =
+                _withdrawRbtcToken(_tokens[i], _maxCheckpoints, _receiver);
+            tempRbtcAmountToSend = tempRbtcAmountToSend.add(totalAmount);
+
+            uint256 _previousUsedCheckpoint = endToken.sub(startingCheckpoint);
+            if (startingCheckpoint > 0) {
+                // we only need to add used checkpoint by 1 only if starting checkpoint > 0
+                _previousUsedCheckpoint.add(1);
+            }
+            _maxCheckpoints = safe32(
+                _maxCheckpoints - _previousUsedCheckpoint,
+                "FeeSharingCollector: maxCheckpoint iteration exceeds 32 bits"
+            );
+        }
+
+        // send all rbtc
+        if (tempRbtcAmountToSend > 0) {
+            (bool success, ) = _receiver.call.value(tempRbtcAmountToSend)("");
+            require(success, "FeeSharingCollector::withdrawRBTC: Withdrawal failed");
+        }
     }
 
     /**
      * @dev Withdraw all of the RBTC balance based starting from a specific checkpoint
      * The function was designed to skip checkpoints with no fees for users
      *
-     * RBTC balance consists of:
+     * This function will withdraw RBTC balance consists of:
      * - rbtc balance
      * - wrbtc balance which will be unwrapped to rbtc
      * - iwrbtc balance which will be unwrapped to rbtc
      *
      * @dev WARNING! This function skips all the checkpoints before '_fromCheckpoint' irreversibly, use with care
      *
-     * @param _fromCheckpoint Skips all the checkpoints before '_fromCheckpoint'
+     * @param _tokens array of rbtc token to be withdrawn
+     * @param _fromCheckpoints Skips all the checkpoints before array of '_fromCheckpoint'
      *        should be calculated offchain with getNextPositiveUserCheckpoint function
      * @param _maxCheckpoints  Maximum number of checkpoints to be processed to workaround block gas limit
      * @param _receiver An optional tokens receiver (msg.sender used if 0)
      */
-    function withdrawRBTCStartingFromCheckpoint(
+    function withdrawRbtcTokensStartingFromCheckpoint(
+        address[] calldata _tokens,
+        uint256[] calldata _fromCheckpoints,
+        uint32 _maxCheckpoints,
+        address _receiver
+    ) external nonReentrant {
+        validFromCheckpointsParam(_fromCheckpoints, _tokens, msg.sender);
+        validRBTCBasedTokens(_tokens);
+
+        if (_receiver == ZERO_ADDRESS) {
+            _receiver = msg.sender;
+        }
+
+        uint256 tempRbtcAmountToSend;
```

Comment:
```suggestion
        uint256 rbtcAmountToSend;
```

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/507/commits/efb8c7eef560dc995d949234265110c21647bded)